shakenfist/baseobject.py

```diff
@@ -111,11 +109,45 @@ def state_filter(states, o):
 
 
 def state_age_filter(delay, o):
-    now = time.time()
-    return (now - o.state.get('state_updated', now)) > delay
+    return (time.time() - o.state.update_time) > delay
 
 
 def namespace_filter(namespace, o):
     if namespace == 'system':
         return True
     return o.namespace == namespace
+
+
+class State(object):
+    def __init__(self, value, update_time, error_msg=None):
+        self.__value = value
+        self.__update_time = update_time
+        self.__error_msg = error_msg
+
+    def __repr__(self):
+        return 'State(' + str(self.json_dump()) + ')'
+
+    def __eq__(self, other):
+        return self.__hash__() == other.__hash__()
+
+    def __hash__(self):
+        return hash(str(self.json_dump()))
+
+    @property
+    def value(self):
+        return self.__value
+
+    @property
+    def update_time(self):
+        return self.__update_time
+
+    @property
+    def error_msg(self):
+        return self.__error_msg
+
+    def json_dump(self):
```

Comment:
Sorry to be _that_guy_, but this isn't JSON right? Its a dictionary you then turn into JSON. I think this should probably be `__dict__`, but I am not 100% sure if that's the intended use of that call or not... Its not completely clear from the exactly one web page I read about it.

Reply:
lol. that_guy.

(I've never tried redefining __dict__() but I fear it will not end well.)

Reply:
Renamed all json_dump()